index.bs

```diff
@@ -3681,11 +3681,44 @@ executes script. The <code>ScriptResultSuccess</code> variant is used in cases
 where the script completes normally and the <code>ScriptResultException</code>
 variant is used in cases where the script completes with a thrown exception.
 
+#### The script.Source type #### {#type-script-Source}
+
+[=Local end definition=]
+
+<pre class="cddl local-cddl">
+Source = (
+  realm: Realm,
+  ?context: BrowsingContext
+);
+</pre>
+
+The <code>Source</code> type represents a <code>script.Realm</code> that
+optionally contains a <code>browsingContext.BrowsingContext</code>.
+
+<div algorithm>
+To <dfn>get the source</dfn> given |source realm|:
+
+1. Let |realm| be the |realm id| for |source realm|.
+
+1. If |source realm| is associated with a [=browsing context=]:
```

Comment:
We don't define anyway what it means for a realm to be "associated with" a browsing context. At the moment you need to go via an environment settings object like

```
1. Let |environment settings| be the [=environment settings object=] whose [=realm execution context=]'s Realm component is realm.

1.  If |environment settings| has a [=responsible document=]:

  1. Let |document| be environment settings’ [=responsible document=].

  1. Let |browsing context| be document’s [=browsing context=].

  1. Let |context id| be the [=browsing context id=] for |browsing context|.
 
  Otherwise let |browsing context| be null.
```

Reply:
Thanks James. I updated the patch and basically used what you added here.